From c9f6e67924075f2c702639de218b355ee405a67a Mon Sep 17 00:00:00 2001
From: Christophe Guerard <christophe.guerard@intel.com>
Date: Mon, 25 Feb 2013 11:43:42 +0100
Subject: [PORT FROM MAIN] dumpstate: fix KW issues

BZ: 89084

Change-Id: If2776b28e0a0ed7b51dc229f24ba5beb83ada0d0
Orig-Change-Id: Ia9b38cadd2c67b79692b6e4c59fefc375af6e61d
Signed-off-by: Christophe Guerard <christophe.guerard@intel.com>
Reviewed-on: http://android.intel.com:8080/93591
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 cmds/dumpstate/dumpstate.c |    8 ++++++--
 cmds/dumpstate/utils.c     |    4 +++-
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/cmds/dumpstate/dumpstate.c b/cmds/dumpstate/dumpstate.c
index 9a332a9..db3f250 100644
--- a/cmds/dumpstate/dumpstate.c
+++ b/cmds/dumpstate/dumpstate.c
@@ -46,6 +46,7 @@ static char screenshot_path[PATH_MAX] = "";
 /* dumps the current system state to stdout */
 static void dumpstate() {
     time_t now = time(NULL);
+    struct tm *time_tmp;
     char build[PROPERTY_VALUE_MAX], fingerprint[PROPERTY_VALUE_MAX];
     char radio[PROPERTY_VALUE_MAX], bootloader[PROPERTY_VALUE_MAX];
     char network[PROPERTY_VALUE_MAX], date[80];
@@ -57,7 +58,8 @@ static void dumpstate() {
     property_get("ro.baseband", radio, "(unknown)");
     property_get("ro.bootloader", bootloader, "(unknown)");
     property_get("gsm.operator.alpha", network, "(unknown)");
-    strftime(date, sizeof(date), "%Y-%m-%d %H:%M:%S", localtime(&now));
+    time_tmp = localtime((const time_t *)&now);
+    strftime(date, sizeof(date), "%Y-%m-%d %H:%M:%S", time_tmp);
 
     printf("========================================================\n");
     printf("== dumpstate: %s\n", date);
@@ -431,7 +433,9 @@ int main(int argc, char *argv[]) {
         if (do_add_date) {
             char date[80];
             time_t now = time(NULL);
-            strftime(date, sizeof(date), "-%Y-%m-%d-%H-%M-%S", localtime(&now));
+            struct tm *time_tmp;
+            time_tmp = localtime((const time_t *)&now);
+            strftime(date, sizeof(date), "-%Y-%m-%d-%H-%M-%S", time_tmp);
             strlcat(path, date, sizeof(path));
         }
         if (do_fb) {
diff --git a/cmds/dumpstate/utils.c b/cmds/dumpstate/utils.c
index 9b0013e..62fa4b0 100644
--- a/cmds/dumpstate/utils.c
+++ b/cmds/dumpstate/utils.c
@@ -226,7 +226,9 @@ int dump_file(const char *title, const char* path) {
         if (memcmp(path, "/proc/", 6) && memcmp(path, "/sys/", 5) && !fstat(fd, &st)) {
             char stamp[80];
             time_t mtime = st.st_mtime;
-            strftime(stamp, sizeof(stamp), "%Y-%m-%d %H:%M:%S", localtime(&mtime));
+            struct tm *time_tmp;
+            time_tmp = localtime((const time_t *)&mtime);
+            strftime(stamp, sizeof(stamp), "%Y-%m-%d %H:%M:%S", time_tmp);
             printf(": %s", stamp);
         }
         printf(") ------\n");
-- 
1.7.4.1

